firebase-storage/ktx/src/main/kotlin/com/google/firebase/storage/ktx/Storage.kt

```diff
@@ -133,6 +140,93 @@ operator fun ListResult.component2(): List<StorageReference> = prefixes
  */
 operator fun ListResult.component3(): String? = pageToken
 
+/**
+ * Starts listening to this task's stream download progress and emits its values via a [Flow].
+ *
+ * - When the returned flow starts being collected, it attaches the following listeners:
+ * [OnPausedListener], [OnProgressListener].
+ * - When the flow completes the listeners will be removed.
+ */
+val StreamDownloadTask.progress: Flow<StorageProgress<StreamDownloadTask.TaskSnapshot>>
+    get() = callbackFlow {
+        val progressListener = OnProgressListener<StreamDownloadTask.TaskSnapshot> { snapshot ->
+            StorageTaskScheduler.getInstance().scheduleCallback {
+                trySendBlocking(StorageProgress.InProgress(snapshot))
+            }
+        }
+        val pauseListener = OnPausedListener<StreamDownloadTask.TaskSnapshot> { snapshot ->
+            StorageTaskScheduler.getInstance().scheduleCallback {
+                trySendBlocking(StorageProgress.Paused(snapshot))
+            }
+        }
+
+        addOnProgressListener(progressListener)
+        addOnPausedListener(pauseListener)
```

Comment:
My initial idea was to pass the same executor to these listeners, but Storage's callback executor is marked as private:
https://github.com/firebase/firebase-android-sdk/blob/4f24be3d2ac6481cb5c3995b4abcf5361894b1c7/firebase-storage/src/main/java/com/google/firebase/storage/StorageTaskScheduler.java#L54-L56

So instead I'm using the `StorageTaskScheduler` singleton to schedule all callbacks in that same executor:
https://github.com/firebase/firebase-android-sdk/blob/27ed605b6503628e96ae9401b26321c53c78a4bc/firebase-storage/ktx/src/main/kotlin/com/google/firebase/storage/ktx/Storage.kt#L153-L155